Source/CarthageKit/Resolver.swift

```diff
@@ -280,6 +280,15 @@ private struct DependencyGraph: Equatable {
 			var nodeSet = edges[dependencyOf] ?? Set()
 			nodeSet.insert(node)
 			edges[dependencyOf] = nodeSet
+
+			// Add a nested dependency to the list of its ancestor.
+			let edgesCopy = edges
+			for (ancestor, var itsDependencies) in edgesCopy {
+				if itsDependencies.contains(dependencyOf) {
+					itsDependencies.insert(node)
+					edges[ancestor] = itsDependencies
+				}
+			}
```

Comment:
I understand, appreciate your input. :pray: 

> `resolver1.edges == resolver2.edges // is this true?`

That was false, I added a fix for the case you pointed out here.